core/app/services/workarea/direct_upload.rb

```diff
@@ -21,7 +18,6 @@ def self.ensure_cors!(request_url)
       cors['CORSConfiguration'].uniq!
 
       Workarea.s3.put_bucket_cors(Configuration::S3.bucket, cors)
```

Comment:
Could compare the cors['CORSConfiguration'] to the original and if changed update the bucket to cut down on unneeded calls.